proposals/array_create.json

```diff
@@ -0,0 +1,65 @@
+{
+    "id": "array_create",
+    "summary": "Create an array",
+    "description": "Creates a new array, which by default is empty.\n\nBy providing the parameter `length`, the array can be pre-filled with the given number of elements. By default, each element is set to `null` unless another value is specified through the parameter `value`.",
+    "categories": [
+        "arrays"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "length",
+            "description": "The number of elements to fill the array with. Defaults to `0`.",
+            "optional": true,
+            "default": 0,
+            "schema": {
+                "type": "integer",
+                "minimum": 0
+            }
+        },
+        {
+            "name": "value",
+            "description": "The value to fill the array with, in case `length` is greater than 0. Defaults to `null` (no data).",
```

Comment:
> I'm wondering if it would be useful/worthwhile to offer more initialization options for the array

I also thought about allowing both single value (repeated x times) and an array of values, but the second use case can also be achieved with array_set: `array_set(data = [], values = [1,2,3])`. So array_create for single value filling, array_set for specific values.

> let user provide another array (as from_node reference)

What is the use case here? As we've figured out "cloning" arrays with non-mutable arrays is not very useful. To duplicate an array you could use array_merge().

Reply:
> What is the use case here?

It's not copying if the desired output length differs from input length, so you need padding or trimming.
Also not if repetition is in play.

Reply:
I think array_create with multiple values would need to be a separate process, otherwise, it's always these "either x or y" descriptions and sometimes parameter apply, sometimes they don't, depending on the other parameter. Also, we can't distinguish whether someone wants to fill each array element with the array given or just use the array as-is. Having array_modify, I think we tackled it mostly and I'm not sure yet how often a "repetition" parameter would actually be used?

Reply:
> Also, we can't distinguish whether someone wants to fill each array element with the array given or just use the array as-is.

good point.

I'm fine with just the basic single value handling